app/src/androidTest/java/com/github/wnder/GuessLocationTourModeActivityTest.java

```diff
@@ -0,0 +1,228 @@
+        intent.putExtra(GuessLocationActivity.EXTRA_GUESS_MODE, R.string.guess_tour_mode);
+        Picture pictureToGuess = new Picture(firstUniqueId, defaultPicLoc.getLatitude(), defaultPicLoc.getLongitude());
+        intent.putExtra(GuessLocationActivity.EXTRA_PICTURE_TO_GUESS, pictureToGuess);
+        intent.putExtra(GuessLocationActivity.EXTRA_TOUR_ID, tourUniqueId);
+    }
+
+    // Clean db
+    @AfterClass
+    public static void afterAll() {
+        deleteTestTour();
+    }
+
+    @Test
+    public void goodThingsAreVisibleWhenZoomingTourMode(){
+        //Before zooming
+        onView(withId(R.id.imageToGuessCard)).check(matches(isDisplayed()));
+        onView(withId(R.id.imageToGuessCardZoomedIn)).check(matches(not(isDisplayed())));
+
+        //Try zooming in
+        onView(withId(R.id.imageToGuessCard)).perform(click());
+        onView(withId(R.id.imageToGuessCard)).check(matches(not(isDisplayed())));
+        onView(withId(R.id.imageToGuessCardZoomedIn)).check(matches(isDisplayed()));
+        onView(withId(R.id.compassMode)).check(matches(not(isDisplayed())));
+        onView(withId(R.id.confirmButton)).check(matches(not(isDisplayed())));
+
+        //Zoom out again
+        onView(withId(R.id.imageToGuessCardZoomedIn)).perform(click());
+        onView(withId(R.id.imageToGuessCard)).check(matches(isDisplayed()));
+        onView(withId(R.id.imageToGuessCardZoomedIn)).check(matches(not(isDisplayed())));
+        onView(withId(R.id.compassMode)).check(matches(not(isDisplayed())));
+        onView(withId(R.id.confirmButton)).check(matches(isDisplayed()));
+    }
+
+    /*
+    @Test
+    public void goThroughTourBySkipping(){
+        onView(withId(R.id.confirmButton)).perform(click());
+        onView(withText(R.string.tour_mode_confirm_while_far_text)).check(matches(isDisplayed()));
+        onView(withText(R.string.tour_mode_confirm_while_far_text)).perform(click());
+
+        onView(withId(R.id.confirmButton)).perform(click());
+        onView(withText(R.string.tour_mode_confirm_while_far_text)).check(matches(isDisplayed()));
+        onView(withId(R.id.backToGuessPreview)).check(matches(not(isDisplayed())));
+        onView(withText(R.string.tour_mode_confirm_while_far_text)).perform(click());
+
+        onView(withId(R.id.backToGuessPreview)).perform(click());
+        onView(withId(R.id.backToGuessPreview)).check(matches(isDisplayed()));
+    }
+
+     */
+
+    /*
+    @Test
+    public void goThroughTourByWalking() {
+        onView(withId(R.id.confirmButton)).perform(click());
+        onView(withId(R.id.confirmButton)).perform(click());
+        onView(withId(R.id.confirmButton)).perform(click());
+
+        onView(withId(R.id.backToGuessPreview)).check(matches(isDisplayed()));
+
+        onView(withId(R.id.backToGuessPreview)).perform(click());
+
+        Intents.intended(hasComponent(GuessPreviewActivity.class.getName()));
+    }
+
+     */

```

Comment:
What do you want to do about those ?

Reply:
Well I guess I should clean up if it doesn't work